compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2703,6 +2722,11 @@ impl<'a> Parser<'a> {
                     enum WrongKw {
                         Duplicated(Span),
                         Misplaced(Span),
+                        /// `MisplacedDisallowedQualifier` is only used instead of `Misplaced`,
+                        /// when the misplaced keyword is disallowed by `is_fn_pointer_type`.
```

Comment:
This is docs on a type but refers to a function parameter which isn't quite clear. Easier to reference when there's an enum type

Reply:
I created the new enum `FrontMatterParsingMode` to solve this